src/test/java/org/jenkinsci/plugins/android_lint/LintWorkflowTest.java

```diff
@@ -0,0 +1,77 @@
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("lint-results.xml");
+        report.copyFrom(LintWorkflowTest.class.getResourceAsStream("/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  step([$class: 'LintPublisher'])\n"
+                        + "}\n", true)
+        );
+        jenkinsRule.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        LintResultAction result = job.getLastBuild().getAction(LintResultAction.class);
+        assertTrue(result.getResult().getAnnotations().size() == 4);
+    }
+
+    /**
+     * Run a workflow job using {@link LintPublisher} with a failing threshold of 0, so the given example file
+     * "/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml" will make the build to fail.
+     */
+    @Test
+    public void lintPublisherWorkflowStepSetLimits() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "lintPublisherWorkflowStepSetLimits");
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("lint-results.xml");
+        report.copyFrom(LintWorkflowTest.class.getResourceAsStream("/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  step([$class: 'LintPublisher', pattern: '**/lint-results.xml', failedTotalAll: '0', usePreviousBuildAsReference: false])\n"
+                        + "}\n", true)
+        );
+        jenkinsRule.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get());
+        LintResultAction result = job.getLastBuild().getAction(LintResultAction.class);
+        assertTrue(result.getResult().getAnnotations().size() == 4);
+    }
+
+    /**
+     * Run a workflow job using {@link LintPublisher} with a unstable threshold of 0, so the given example file
+     * "/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml" will make the build to fail.
+     */
+    @Test
+    public void lintPublisherWorkflowStepFailure() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "lintPublisherWorkflowStepFailure");
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("lint-results.xml");
+        report.copyFrom(LintWorkflowTest.class.getResourceAsStream("/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  step([$class: 'LintPublisher', pattern: '**/lint-results.xml', unstableTotalAll: '0', usePreviousBuildAsReference: false])\n"

```

Comment:
There should probably be another test that uses something other than the default `**/lint-results.xml` value, otherwise you're never actually verifying that the `pattern` parameter works?